### PR TITLE
Refactor spatial unit tests.

### PR DESCRIPTION
**Pull request by Vilin97**
Refactoring the tests for the spatial SSAs. Adding more comprehensive unit tests.

### CODE REVIEW

**Comment by isaacsas** [comment]
Thanks for taking a pass on these!

File: test/spatial/topology.jl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+# Functions to test:
+# num_sites(grid)
+# neighbors(grid, site)
+# outdegree(grid, site)
+# rand_nbr(rng, grid, site)
+# nth_nbr(grid, site, n)
+
+io = IOBuffer()
+rng = MersenneTwister()
```
Comment (isaacsas) [inline test/spatial/topology.jl:13]: Use `StableRng`s with a fixed seed, see the other tests.

**Comment by Vilin97** [comment]
@isaacsas could you please explain how to run the formatter? https://github.com/SciML/SciMLStyle#juliaformatter is either out of date or I am doing something wrong. Getting `ERROR: AssertionError: currently .JuliaFormatter.toml accepts only "default" or "yas" or "blue" for the style configuration `

**Comment by isaacsas** [comment]
Sounds like you may have an older version of JuliaFormatter before the style was added?

File: test/spatial/hop_rates.jl [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    end
+    for species in 1:num_species
+        @test abs(spec_dict[species]) / num_samples - spec_probs[species] < rel_tol
+    end
+    for target in JP.neighbors(g, site)
+        @test abs(site_dict[target]) / num_samples - target_probs[target] < rel_tol
+    end
+end
+
+io = IOBuffer()
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:50]:
```suggestion

```

File: test/spatial/hop_rates.jl [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+g = CartesianGridRej(dims)
+num_nodes = JP.num_sites(g)
+num_species = 2
+u = ones(Int, num_species, num_nodes)
+
+# Tests for HopRatesGraphDs
+hop_constants = [1.0, 3.0] # species
+hop_rates = JP.HopRates(hop_constants, g)
+@test hop_rates isa JP.HopRatesGraphDs
+show(io, "text/plain", hop_rates)
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:64]:
```suggestion

```

File: test/spatial/hop_rates.jl [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    end
+    statistical_test(hop_rates, spec_propensities, target_propensities, num_species, u,
+                     site, g, rng, rel_tol)
+end
+test_reset(hop_rates, num_nodes)
+
+# Tests for HopRatesGraphDsi
+hop_constants = [2.0 1.0 1.0 1.0 1.0 1.0; 6.0 3.0 3.0 3.0 3.0 3.0] # [species, site]
+hop_rates = JP.HopRates(hop_constants, g)
+@test hop_rates isa JP.HopRatesGraphDsi
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:79]:
```suggestion

```

File: test/spatial/hop_rates.jl [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+show(io, "text/plain", hop_rates)
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:80]:
```suggestion

```

File: test/spatial/hop_rates.jl [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    [3.0, 6.0, 12.0],
+    [3.0, 6.0],
+]
+hop_rates_structs = [
+    JP.HopRatesGraphDsij(hop_constants),
+    JP.HopRates(hop_constants, g),
+]
+@test hop_rates_structs[2] isa JP.HopRatesGridDsij
+for hop_rates in hop_rates_structs
+    show(io, "text/plain", hop_rates)
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:116]:
```suggestion

```

File: test/spatial/hop_rates.jl [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+    [1.0, 2.0, 4.0],
+    [1.0, 2.0],
+] #[site][target_site]
+hop_rates_structs = [
+    JP.HopRatesGraphDsLij(species_hop_constants, site_hop_constants),
+    JP.HopRates((species_hop_constants => site_hop_constants), g),
+]
+@test hop_rates_structs[2] isa JP.HopRatesGridDsLij
+for hop_rates in hop_rates_structs
+    show(io, "text/plain", hop_rates)
```
Comment (isaacsas) [inline test/spatial/hop_rates.jl:146]:
```suggestion

```

**Review by isaacsas — APPROVED** [review]

**Comment by isaacsas** [comment]
Thanks!

**Pull request merged**